pkg/controller/directvolumemigration/rsync.go

```diff
@@ -76,16 +76,35 @@ func (t *Task) ensureRsyncEndpoint() error {
 	dvmLabels["purpose"] = DirectVolumeMigrationRsync
 	for bothNs := range t.getPVCNamespaceMap() {
 		ns := getDestNs(bothNs)
+
+		// Get cluster subdomain if it exists
+		cluster, err := t.Owner.GetDestinationCluster(t.Client)
+		if err != nil {
+			return err
+		}
+
+		// Get the user provided subdomain, if empty we'll attempt to
+		// get the cluster's subdomain from the cluster's ingress config
+		subdomain, err := cluster.GetClusterSubdomain(t.Client)
+		if err != nil {
+			t.Log.Info("failed to get cluster_subdomain" + err.Error() + "attempting to get cluster's ingress domain")
+			subdomain, err = cluster.GetClusterIngressDomain(t.Client)
+			if err != nil {
+				t.Log.Error(err, "failed to retrieve cluster's ingress domain, extremely long namespace names will cause route creation failure")
```

Comment:
Not sure if I should log this as an error or not.

Reply:
+1 to logging, might even want to put it as a warning condition?. Also does the destClient have ingressConfig scheme registered for this to work?